cmd/ntpcheck/cmd/utils.go

```diff
@@ -108,13 +124,29 @@ func ntpDate(remoteServerAddr string, remoteServerPort string, requests int) err
 		return err
 	}
 
-	fmt.Printf("Server: %s, Requests: %d\n", addr, requests)
-	var sumAvgNetworkDelay int64
+	var sumDelay int64
 	var sumOffset int64
+	var clientTransmitTime time.Time
+	var actualTransmitTime time.Time
 
 	for i := 0; i < requests; i++ {
-		clientTransmitTime := time.Now()
+		actualTransmitTime = time.Now()
+		if clientTransmitOffset != 0 {
+			clientTransmitTime = actualTransmitTime.Add(clientTransmitOffset)
+		} else if userDate != "" {
+			clientTransmitTime, err = dateparse.ParseLocal(userDate)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+		} else {
+			clientTransmitTime = actualTransmitTime
+		}
+		log.Debugf("Client TX timestamp (system): %v\n", clientTransmitTime)
+
 		sec, frac := ntp.Time(clientTransmitTime)
+		clientWireTransmitTime := ntp.Unix(sec, frac)
```

Comment:
Wait why are you converting it back and forth?

Reply:
This shows when the system and NTP times don't match due to NTP era start/end e.g.
```
ntpcheck utils ntpdate -s time4.facebook.com -d "2/7/2036 12:28:16 UTC" -v
DEBU[0000] Client TX timestamp (system): 2036-02-07 12:28:16 +0000 UTC
DEBU[0000] Client TX timestamp (ntp): 1899-12-31 22:00:00 -0800 PST
```

Reply:
Sorry maybe I was not clear - `clientTransmitTime` (time) and `clientWireTransmitTime` (time) should be the same. If you want to test it - maybe unittests is a better option.

Reply:
This is the protocol acting as designed. This can occur when system time is encoded as NTP timestamp (uint64).